### PR TITLE
Separate form and search results (Part 2 of 2) 

### PR DESCRIPTION
**Pull request by takshch**
# Description
- Removed queryParams from index route.
- Added queryParams to changes route.
- No change in url of index route when we select or change ember version (fromVersion, toVersion).
- When click on the 'Find Changes' button, we will be redirected to changes route with queryParams.

# Acceptance test for index route
- When we visit the index route, we see the form for selecting Ember versions ✅
- When we select Ember versions, the URL does not change ✅
- When we submit the form, we are redirected to the changes route ✅
- When we submit the form, we see the features and deprecations that occurred ✅
- When we submit the form with fromVersion greater than toVersion, we see 0 features and 0 deprecations ✅

### CODE REVIEW

**Comment by ijlee2** [comment]
Thanks, @takshch. I'll have a look later tonight.

File: app/controllers/index.js [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
   @action displayChanges({ fromVersion, toVersion }) {
     this.fromVersion = fromVersion;
     this.toVersion = toVersion;
+    this.router.transitionTo('changes', {
+      queryParams: {
+        fromVersion: this.fromVersion,
+        toVersion: this.toVersion,
+      },
+    });
```
Comment (ijlee2) [inline app/controllers/index.js:16]: **suggestion (soft-blocking):**

The controller doesn't need its copy of `fromVersion` and `toVersion`, now that we have a form component. Let's delete lines 9 and 10, and update lines 11-16 to:

```suggestion
    this.router.transitionTo('changes', {
      queryParams: {
        fromVersion,
        toVersion,
      },
    });
```

File: tests/acceptance/index-test.js [hunk 2]
```diff
@@ -161,4 +173,32 @@ module('Acceptance | index', function (hooks) {
       version: '3.2',
     });
   });
+
+  test('When we submit the form with fromVersion greater than toVersion, we see 0 features and 0 deprecations', async function (assert) {
```
Comment (ijlee2) [inline tests/acceptance/index-test.js:177]: **praise:**

Thanks for checking this edge case!

**Review by ijlee2 — CHANGES_REQUESTED** [review]
@takshch Thanks for working on this issue! I checked the app locally and everything worked well. I thought you did an exceptional job with writing tests. ✨

Before I merge your pull request, I'd like to see a couple of changes:

- Remove references to `this.fromVersion` and `this.toVersion` from the `index` controller
- In the `index` route template, can you remove the "form's output" (lines 7-39)? (The user is to see the output in `changes` route now.)

**Comment by takshch** [comment]
Required changes are made and pushed into the code. Please review it Sir.

File: app/controllers/index.js (hunk 1 above)
Comment (takshch) [reply app/controllers/index.js]: Yes. It's mistake. and it's fixed now.

**Comment by takshch** [comment]
@ijlee2 Sorry for unexpected hbs linting changes. Could you please tell me when to  and when not to use hbs linter.

**Comment by ijlee2** [comment]
@takshch No problem. It seemed like you may have an IDE that auto-fixes linting errors.

For pull requests for open source, unless the issue asked you to fix linting errors, I'd try to preserve the code formatting (make few code changes to help reduce noise in a code review).

**Review by ijlee2 — APPROVED** [review]

**Comment by takshch** [comment]
@ijlee2 Could you please add label of "hacktoberfest-accepted"

**Comment by ijlee2** [comment]
Hi, @takshch. If you visit your [Hacktoberfest dashboard](https://hacktoberfest.digitalocean.com/profile), I think you will see a timer next to this pull request.

The [updated rules](https://hacktoberfest.digitalocean.com/hacktoberfest-update) indicate that there's a **14-day waiting period** before you see a checkmark next to the pull request. Because I already added `Hacktoberfest` topic to this repo and approved your pull request, you should be good to go. 🎊

updated_rules
